luavm/src/lib/instructions/functions.rs

```diff
@@ -15,15 +15,33 @@ pub fn closure(vm: &mut Vm, instr: u32) -> Result<(), LuaError> {
 }
 
 pub fn push(vm: &mut Vm, instr: u32) -> Result<(), LuaError> {
-    vm.stack
-        .push(vm.registers[first_arg(instr) as usize].clone());
+    // push all the variable arguments of the current function to the stack
+    if second_arg(instr) == 1 {
+        let param_count = vm.bytecode.get_function(vm.curr_func).param_count();
+        let args_count = vm.closure.closure_args_count()?;
+        if args_count > param_count {
+            // make sure to skip the arguments which are the actual parameters
+            let mut var_args_start = vm.closure.closure_args_start()? + param_count;
```

Comment:
I think this would be clearer as something like:

```rust
for i in vm.closure.closure_args_start() + param_count .. args_count - param_count {
 ...
}
let var_args_start = args_count - param_count
```

[Warning: I might not have used the right offsets.]